lib/api_auth/request_drivers/faraday.rb

```diff
@@ -56,7 +56,9 @@ def content_md5
       end
 
       def request_uri
-        uri = URI::HTTP.new(nil, nil, nil, nil, nil, @request.path, nil, @request.params.to_query, nil)
+        query_string = @request.params.map{|k,v| "#{k}=#{v}"}.join('&')
```

Comment:
to_query will sort the query which is incorrect. For instance if we go to /resource.xml?foo=bar&bar=foo the server will build it's canonical string with the parameters in that order while the client will have them sorted when it signs it. So the signatures won't match.

Unfortunately this is the exact same problem we are getting with my new solution in Ruby pre 1.9. It wasn't until Ruby 1.9 the ordering in hashes was specified (http://stackoverflow.com/a/17355411)

So the fact that these parameters are stored in a hash in Faraday and that is the only way to build the URI means that it may never work in 1.8.7 because we can never be guaranteed to know what order those parameters were in when the request was made. Faraday obviously stores the original string used to create the params list because it keeps the original non-alphabetical order when it makes the request.